cartridges/int_klaviyo_core/cartridge/scripts/klaviyo/utils.js

```diff
@@ -222,6 +222,87 @@ function trackEvent(exchangeID, data, event, customerEmail) {
 }
 
 
+// The subscribeUser func takes the user email & phone number to prep a data object w/ a corresponding emailListID or smsListID (both configured in BM w/ values from the Klaviyo Dashboard)
+// Data is sent to the KlaviyoSubscribeProfilesService API to subscribe users to email or SMS lists.
+function subscribeUser(subscribeToEmail, subscribeToSMS, email, phone) {
+    var logger = Logger.getLogger('Klaviyo', 'Klaviyo.core utils.js - subscribeUser()');
+
+    if (klaviyoServices.KlaviyoSubscribeProfilesService == null) {
+        logger.error('subscribeUser() failed - KlaviyoSubscribeProfilesService is null.');
+        return;
+    }
+
+    var emailListID = Site.getCurrent().getCustomPreferenceValue('klaviyo_email_list_id');
+    var smsListID = Site.getCurrent().getCustomPreferenceValue('klaviyo_sms_list_id');
+
+    var data;
+    var result;
+
+    if (subscribeToEmail && emailListID) {
+        data = {
+            data: {
+                type       : 'profile-subscription-bulk-create-job',
+                attributes : {
+                    list_id       : emailListID,
+                    custom_source : 'Marketing Event',
+                    subscriptions : [{
+                        channels     : { email: ['MARKETING'] },
+                        email        : email,
+                        phone_number : phone
+                    }]
+                }
+            }
+        };
+
+        result = klaviyoServices.KlaviyoSubscribeProfilesService.call(data);
+
+        if (result == null) {
+            logger.error('klaviyoServices.KlaviyoSubscribeProfilesService subscribe call for email returned null result');
+        }
+
+        if (!result.ok === true) {
+            logger.error('klaviyoServices.KlaviyoSubscribeProfilesService subscribe call for email error: ' + result.errorMessage);
+            // check to see if the reason the call failed was because of Klaviyo's internal phone number validation.  if so, resend without phone number
+            var errObj = JSON.parse(result.errorMessage);
+            if (result.error == 400 && errObj.errors[0].source.pointer == 'phone_number') {
+                data.data.attributes.subscriptions[0].phone_number = null;
+                result = klaviyoServices.KlaviyoSubscribeProfilesService.call(data);
+                if (result == null) {
```

Comment:
will this ever happen if the `result.error` and `result.errorMessage` are not `null`?